nav2_planner/src/planner_server.cpp

```diff
@@ -510,7 +517,72 @@ PlannerServer::publishPlan(const nav_msgs::msg::Path & path)
   }
 }
 
-}  // namespace nav2_planner
+void PlannerServer::isPathValid(
+  const std::shared_ptr<nav2_msgs::srv::IsPathValid::Request> request,
+  std::shared_ptr<nav2_msgs::srv::IsPathValid::Response> response)
+{
+  unsigned int mx = 0;
+  unsigned int my = 0;
+
+  response->is_valid = true;
+
+  if (request->path.poses.empty() ) {
+    response->is_valid = false;
+  } else {
+    geometry_msgs::msg::PoseStamped current_pose;
+    unsigned int closest_point_index = 0;
+    if (costmap_ros_->getRobotPose(current_pose)) {
+      float closest_distance = std::numeric_limits<float>::max();
+      float current_distance = std::numeric_limits<float>::max();
+      float prev_distance = std::numeric_limits<float>::max();
+      unsigned int increasing_distance_counter = 0;
+      const unsigned int MAX_INCREASE = 3;
+      geometry_msgs::msg::Point current_point;
+      current_point = current_pose.pose.position;
+      for (unsigned int i = 0; i < request->path.poses.size(); ++i) {
+        geometry_msgs::msg::Point path_point;
+        path_point.x = request->path.poses[i].pose.position.x;
```

Comment:
`request->path.poses[i].pose.position` is a `geometry_msgs::msg::Point`, you can just do

`geometry_msgs::msg::Point path_point = request->path.poses[i].pose.position`

Reply:
Done